### DISCUSSION TITLE
Stanc3 online grammar editor/tester

### DISCUSSION THREAD

**Discussion by mingodad** [stan-dev/stanc3#1362] [Show and tell]
I've just added this project grammar to https://mingodad.github.io/parsertl-playground/playground/ (select `Stanc3 parser` under examples then click `Parse` to see the parser tree).

The main repository is here https://github.com/mingodad/parsertl-playground it's a `Yacc/Lex` runtime grammar evaluator that can provide quick feedback loop when developing/testing/debug grammars.

It can also generate  `EBNF` compatible with https://www.bottlecaps.de/rr/ui to generate railroad diagrams.

I hope it can help develop/debug/extend/test/document the grammar used in this project.

Any feedback is welcome !

**Comment by WardBrian** [comment]
Cool!

Does this automatically update as `stanc3` evolves? Or is it manual?
In either case, it might be nice to give a version or git hash of the parser being used

**Comment by mingodad** [comment]
It need manual update.
I just updated the converted grammar to include the version info:
```
/*
From:
	https://github.com/stan-dev/stanc3/blob/2da1bbc75f544885b1f5de9bf98f027911c83f1e/src/frontend/parser.mly
	https://github.com/stan-dev/stanc3/blob/2da1bbc75f544885b1f5de9bf98f027911c83f1e/src/frontend/lexer.mll
menhir --only-preprocess-for-ocamlyacc parser.mly > parser.mly.yacc
ocamlyacc -v parser.mly.yacc
manual extraction and fixes including lexer.mll
*/
```
